### PR TITLE
Debug standalone build

### PR DESCRIPTION
**Pull request by rosteen**
Working on getting the binary build working in CI again. With these changes I have the build process working on my local machine (Mac), but it's still failing on CI because the page playwright is getting in the test is a `500: Internal Server Error` page. This is surely just hiding some other error that's getting hit running the notebook, but I've resolved all the errors I was seeing locally so I'm not sure where to go next with this.

This is in draft because there are still a few changes in here that I was just using to debug (e.g.  the `pysiaf`-related print statements and the extra chromium install).

### CODE REVIEW

**Comment by rosteen** [comment]
I marked this as not needing a changelog since it's not user-facing.

File: .github/workflows/standalone.yml
Comment (kecnry) [inline .github/workflows/standalone.yml:7]: do we want to keep this or remove it?  Or maybe some other wildcard prefix so we don't have to edit main in the future?

File: standalone/jdaviz-cli-entrypoint.py
Comment (kecnry) [inline standalone/jdaviz-cli-entrypoint.py:2]: we'll need to decide what to do with this (and `hook-pysiaf`) given the plan to keep as a soft-dependency for now....

**Comment by kecnry** [comment]
> I marked this as not needing a changelog since it's not user-facing.

Will merging this PR mean that 3.7 will ship with a standalone executable?  Are we ready for that?  And if so - maybe that is deserving of a changelog?

File: .github/workflows/standalone.yml
Comment (pllim) [reply .github/workflows/standalone.yml:7]: Is this branch to stay? I thought it was a temporary branch.

File: standalone/hooks/hook-pysiaf.py
Comment (pllim) [inline standalone/hooks/hook-pysiaf.py:3]: What does this mean? pysiaf is not optional anymore?

File: standalone/jdaviz-cli-entrypoint.py
Comment (pllim) [reply standalone/jdaviz-cli-entrypoint.py:2]: What is this import for anyway? I don't see it used here.

File: standalone/test.py
Comment (pllim) [inline standalone/test.py:1]: Is the filename required to be "test.py" by pyinstaller? If not, can the filename be a bit more descriptive?

**Review by pllim — COMMENTED** [review]
Thanks! 🍺

**Comment by pllim** [comment]
devdeps and RTD failures are unrelated.

File: .github/workflows/standalone.yml
Comment (pllim) [reply .github/workflows/standalone.yml:7]: Also, do we want the option to enable in PR using special label, or not worth it?

File: standalone/hooks/hook-pysiaf.py
Comment (rosteen) [reply standalone/hooks/hook-pysiaf.py:3]: There's no way we can make it optional in the standalone executable, given that all the dependencies are packaged together by pyinstaller. We either include it or we don't - I can't remember if not including it led to an error in the standalone build, but I figure we should have the full functionality available in the executable version?

Comment (rosteen) [reply standalone/hooks/hook-pysiaf.py:3]: It's still optional if you install from pypi or from source.

**Comment by rosteen** [comment]
> > I marked this as not needing a changelog since it's not user-facing.
> 
> Will merging this PR mean that 3.7 will ship with a standalone executable? Are we ready for that? And if so - maybe that is deserving of a changelog?

Define "ship with". The executables will exist in the CI build jobs on main, but that doesn't mean that we have to tell any users about them or put them anywhere more accessible for users to download. I guess the answer to your question is "no". It's not like users will automatically get these executables if they `pip install jdaviz` or something.

**Comment by pllim** [comment]
> not like users will automatically get these executables

So... how does these tie into the release process going forward?

**Comment by rosteen** [comment]
> > not like users will automatically get these executables
> 
> So... how does these tie into the release process going forward?

TBD 😅

**Comment by kecnry** [comment]
ok, good to know.  So this is strictly testing at this point.  And at some point in the future when we want to consider the standalone build to be official/stable, then we will need to work it into release procedures so its packaged along with the release?

**Comment by rosteen** [comment]
Longer answer: I don't think we've made a decision on where/how to distribute these. But this PR will at least make our builds stop failing in the meantime. I do think we should decide how to distribute before we release 3.7, but that doesn't need to hold up this PR.

**Review by kecnry — APPROVED** [review]
changes look reasonable once CI is happy and the branch override is then removed from the workflow (which will then likely break CI, so we'll need to override to actually merge).

**Comment by rosteen** [comment]
I removed this branch from the workflows file now that I saw the build succeed with the last changes. I think this is ok to merge now.

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

Should open follow-up issue about the release and milestone it to 3.7.

**Comment by kecnry** [comment]
(a squash is probably warranted, this is a lot of debugging commits)

**Comment by pllim** [comment]
Hmm... I can always make the only option to be squash merge on repo settings... 🤔

**Comment by pllim** [comment]
RTD is going to keep failing until you rebase. But I don't think you have to since this PR doesn't touch docs. Ricky or I can override the rules.

**Comment by rosteen** [comment]
See #2405 for the follow-up issue. I'll override and merge (and I was definitely planning to squash this!).

**Comment by pllim** [comment]
Now it fails on `main` under OSX, complaining about "Unnotarized Developer ID".

Example log: https://github.com/spacetelescope/jdaviz/actions/runs/6017274758/job/16323057660

**Pull request merged**
